app/src/main/java/com/jrod7938/textchangeapp/components/Components.kt

```diff
@@ -1,1160 +1,1582 @@
+                onClick = onItemClick,
+            )
+            // VerticalDivider()
+        }
+        SelectionPill(
+            option = last,
+            selected = state.contains(last.text),
+            onClick = onItemClick,
+        )
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun BookThumbnail(
+    book: MBook,
+    viewModel: BookInfoScreenViewModel = viewModel(),
+    navController: NavHostController,
+    ){
+
+    val user by viewModel.user.observeAsState(initial = null)
+
+    LaunchedEffect(true) { viewModel.getUser() }
+
+    val (isChecked, setChecked) = remember { mutableStateOf(false) }
+    val (view, setView) = remember { mutableStateOf(false)}
+
+    val context = LocalContext.current
+
+    Column(
+        verticalArrangement = Arrangement.Center,
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(30.dp)) {
+        Image(
+            painter = rememberAsyncImagePainter(model = book.imageURL),
+            contentDescription = "Image of ${book.title}",
+            modifier = Modifier.size(175.dp)
+        )
+
+        Text(
+            text = book.title,
+            fontWeight = FontWeight.Bold,
+            color = MaterialTheme.colorScheme.primary,
+            maxLines = 1,
+            modifier = Modifier.padding(top = 20.dp)
+        )
+
+        Text(text = "by ${book.author}")
+
+
+        Text(
+            text = "Price: $${book.price}",
+            color = MaterialTheme.colorScheme.secondary,
+        )
+        Text(text = "Condition: ${book.condition}")
+        Column() {
+            Row(verticalAlignment = Alignment.CenterVertically,
+                horizontalArrangement = Arrangement.Center,
+                modifier = Modifier.fillMaxWidth()) {
+                SavedToFavoritesButton(
+                    isChecked = isChecked,
+                    onClick = {
+                        if(user?.savedBooks?.contains(book.bookID)!!){
+                            viewModel.unsaveBook(book)
+                            Toast.makeText(
+                                context,
+                                "Removed from Saved",
+                                Toast.LENGTH_SHORT
+                            ).show()
+
+                        } else {
+                            viewModel.saveBook(book)
+                            Toast.makeText(
+                                context,
+                                "Added to Saved",
+                                Toast.LENGTH_SHORT
+                            ).show()
+                        }
+                            setChecked(!isChecked)
+                    })
+                Button(
+                    onClick = { setView(true) }
+                ) {
+                    Text(text = "Purchase")
+                }
+                Icon(
+                    Icons.Default.MoreHoriz,

```

Comment:
My system does not recognize Icons.Default.MoreHoriz, please ensure this Icon is correct

Reply:
This is correct. Did you sync the grade project files?

Reply:
Sync fixed the issue